src/vtl.ts

```diff
@@ -99,11 +99,19 @@ export class VTL {
    * @param expr should be a quoted string or a variable that represents the value to set
    */
   public put(objVar: string, name: string, expr: string | Expr) {
-    this.qr(
-      `${objVar}.put(${name}, ${
-        typeof expr === "string" ? expr : this.eval(expr)
-      })`
-    );
+    let result;
+    if (typeof expr === "string") {
+      result = expr;
+    } else if (isBinaryExpr(expr)) {
```

Comment:
By the way, I think it's possible for this BinaryExpr to be an assignment where the operator is `=`

Are we handling this right?

Reply:
Wouldn't that be a `VariableStmt`?

Reply:
No, it would be an ExprtStmt(BinaryExpr) if it's assigning without const/let/var

```
a = b
```

Reply:
It's an expression that returns the value that is assigned. 

```ts
this.a = (a = b)
```

Reply:
Gross. You have to go out of your way to make it happen, but we are not handling it correctly.

```
// @ts-ignore
let y = 1;
return { x: (y = 2) };
```

Will fix in another PR